src/reader.rs

```diff
@@ -1449,13 +1455,21 @@ impl<R: io::Read> Reader<R> {
         // byte headers, then get the string headers (or a UTF-8 error).
         let (str_headers, byte_headers) = match headers {
             Ok(string) => {
-                let bytes = string.clone().into_byte_record();
+                let mut bytes = string.clone().into_byte_record();
+                match self.core.trim {
```

Comment:
I think it would be better to define (unexported) predicates on the `Trim` type. e.g., `should_trim_headers` and `should_trim_field`.